### PR TITLE
[APPWIZ] Make shortcut name selected on creation

### PR DESCRIPTION
**Pull request by zecarlos1957**
Show editbox text in selected mode (in blue);

### CODE REVIEW

File: dll/cpl/appwiz/createlink.c [hunk 1]
```diff
@@ -265,7 +265,7 @@ WelcomeDlgProc(HWND hwndDlg,
                     PathRemoveBackslashW(pContext->szWorkingDirectory);
                     pch = PathFindFileNameW(pContext->szWorkingDirectory);
                     if (pch && *pch)
-                        *pch = 0;
```
Comment (ThFabba) [inline dll/cpl/appwiz/createlink.c]: Copying the file name into `szDescription` makes sense. But why are you removing the `*pch = 0`, which removes the file name from `szWorkingDirectory`?

Comment (zecarlos1957) [reply dll/cpl/appwiz/createlink.c]: Yes! You're righ.

**Comment by zecarlos1957** [comment]
The three instructions on lines 262, 263, and 264 (modified file), appear to be of no use at all, as we are about to leave the current page and do not need the focus on IDC_SHORTCUT_LOCATION, nor select its text. Would it make sense to delete these three instructions?

File: dll/cpl/appwiz/createlink.c [hunk 2]
```diff
@@ -227,7 +227,11 @@ WelcomeDlgProc(HWND hwndDlg,
                         break;
 
                     if (SHGetPathFromIDListW(pidllist, szPath))
+                    {
                         SetDlgItemTextW(hwndDlg, IDC_SHORTCUT_LOCATION, szPath);
+                        SendDlgItemMessageW(hwndDlg, IDC_SHORTCUT_LOCATION, WM_SETFOCUS, 0, 0);
+                        SendDlgItemMessageW(hwndDlg, IDC_SHORTCUT_LOCATION, EM_SETSEL, 0, -1);
+                    }
```
Comment (HBelusca) [inline dll/cpl/appwiz/createlink.c]: OK (keep it)

File: dll/cpl/appwiz/createlink.c [hunk 3]
```diff
@@ -255,8 +260,6 @@ WelcomeDlgProc(HWND hwndDlg,
                 else if (GetFileAttributesW(pContext->szTarget) != INVALID_FILE_ATTRIBUTES)
                 {
                     /* file */
-                    SendDlgItemMessage(hwndDlg, IDC_SHORTCUT_LOCATION, EM_SETSEL, 0, -1);
-                    SetFocus(GetDlgItem(hwndDlg, IDC_SHORTCUT_LOCATION));
```
Comment (HBelusca) [inline dll/cpl/appwiz/createlink.c]: May be OK.

File: dll/cpl/appwiz/createlink.c [hunk 4]
```diff
@@ -240,6 +244,7 @@ WelcomeDlgProc(HWND hwndDlg,
             {
                 LPWSTR pch;
                 pContext = (PCREATE_LINK_CONTEXT) GetWindowLongPtr(hwndDlg, DWLP_USER);
+                PropSheet_SetWizButtons(GetParent(hwndDlg), PSWIZB_BACK | PSWIZB_FINISH);
```
Comment (HBelusca) [inline dll/cpl/appwiz/createlink.c]: I don't think this is correct.

File: dll/cpl/appwiz/createlink.c [hunk 5]
```diff
@@ -278,6 +284,11 @@ WelcomeDlgProc(HWND hwndDlg,
                     MessageBoxW(hwndDlg, szError, szDesc, MB_ICONERROR);
                 }
             }
+            else if (lppsn->hdr.code == PSN_QUERYCANCEL)
+            {
+                pContext = (PCREATE_LINK_CONTEXT)GetWindowLongPtr(hwndDlg, DWLP_USER);
+                DeleteFileW(pContext->szOldFile);
+            }
```
Comment (HBelusca) [inline dll/cpl/appwiz/createlink.c]: Should not be necessary anymore.

File: dll/cpl/appwiz/createlink.c [hunk 6]
```diff
@@ -301,6 +312,7 @@ FinishDlgProc(HWND hwndDlg,
             pContext = (PCREATE_LINK_CONTEXT) ppsp->lParam;
             SetWindowLongPtr(hwndDlg, DWLP_USER, (LONG_PTR)pContext);
             SetDlgItemTextW(hwndDlg, IDC_SHORTCUT_NAME, pContext->szDescription);
+            SendDlgItemMessage(hwndDlg, IDC_SHORTCUT_NAME, EM_SETSEL, 0, -1);
```
Comment (HBelusca) [inline dll/cpl/appwiz/createlink.c]: Is this still needed?

File: dll/cpl/appwiz/createlink.c [hunk 7]
```diff
@@ -376,6 +388,15 @@ FinishDlgProc(HWND hwndDlg,
                     }
                 }
             }
+            else if (lppsn->hdr.code == PSN_WIZBACK)
+            {
+                PropSheet_SetWizButtons(GetParent(hwndDlg), PSWIZB_NEXT);
+            }
+            else if (lppsn->hdr.code == PSN_QUERYCANCEL)
```
Comment (HBelusca) [inline dll/cpl/appwiz/createlink.c]: This PSN_QUERYCANCEL is not needed anymore.

File: dll/cpl/appwiz/createlink.c [hunk 8]
```diff
@@ -376,6 +388,15 @@ FinishDlgProc(HWND hwndDlg,
                     }
                 }
             }
+            else if (lppsn->hdr.code == PSN_WIZBACK)
+            {
+                PropSheet_SetWizButtons(GetParent(hwndDlg), PSWIZB_NEXT);
```
Comment (HBelusca) [inline dll/cpl/appwiz/createlink.c]: Setting the wizard buttons for the current page must be done by that page for consistency purposes.

**Comment by KyleKatarn31** [comment]
What is still needed for this PR ?
Would you please post a pre/post modification screenshots please ?

**Comment by zecarlos1957** [comment]
before
After
Just waiting from approval

**Comment by katahiromz** [comment]
Please rename the PR title.

**Pull request merged**